builtin/logical/pki/cert_util.go

```diff
@@ -80,9 +80,32 @@ func getFormat(data *framework.FieldData) string {
 	return format
 }
 
-// fetchCAInfo will fetch the CA info, will return an error if no ca info exists.
+// fetchCAInfo will fetch the CA info, will return an error if no ca info exists, this does NOT support
+// loading using the legacyBundleShimID and should be used with care. This should be called only once
+// within the request path otherwise you run the risk of a race condition with the issuer migration on perf-secondaries.
 func fetchCAInfo(ctx context.Context, b *backend, req *logical.Request, issuerRef string, usage issuerUsage) (*certutil.CAInfoBundle, error) {
-	entry, bundle, err := fetchCertBundle(ctx, b, req.Storage, issuerRef)
+	var issuerId issuerID
```

Comment:
Remind me again why it wasn't possible to use `fetchCertBundle` as the flex point again, rather than adding this here? It seems like the two would be equivalent: there's a single check for `useLegacyBundleCaStore` in both, and we should never go backwards (from `true->false`) on that variable. 

About the only material change I think I can see is two places that otherwise handled this fine now just directly call the `fetchCAInfoByIssuerId` / `fetchCertBundleByIssuerId` and the legacy bundle shim now is better which I _think_ is the core of the change to make CRL building work.

Maybe I'm misunderstanding that though? Regardless, I think I'm fine with it once I've wrapped my head around it.

Reply:
Ah, ok, that's fair :-) Sounds good to me then!